app/assets/javascripts/controllers/container_deployment/deploy-provider-controller.js

```diff
@@ -0,0 +1,254 @@
+            dns: item.dns,
+            etcd: item.etcd,
+            infrastructure: item.infrastructure
+          }
+        });
+      });
+      nodes = nodes.filter(function (node) {
+        return node.roles.master || node.roles.node || node.roles.storage
+      });
+      return nodes;
+    };
+
+    var create_deployment_resource = function() {
+      var method_types = {
+        existingVms: "existing_managed",
+        newVms: "provision",
+        noProvider: "unmanaged"
+      };
+
+      var resource = {
+        provider_name: $scope.data.providerName,
+        provider_type: $scope.data.providerType,
+        method_type: method_types[$scope.data.provisionOn],
+        rhsm_authentication: {
+          mode: "AuthenticationRhsm",
+          userid: $scope.data.rhnUsername,
+          password: $scope.data.rhnPassword,
+          rhsm_sku: $scope.data.rhnSKU,
+          rhsm_server: $scope.data.rhnSatelliteUrl
+        },
+        ssh_authentication: {
+          mode: "AuthPrivateKey",
+          userid: $scope.data.deploymentUsername,
+          auth_key: $scope.data.deploymentKey
+        },
+        nodes: create_nodes_object(),
+        identity_authentication: create_auth_object()
+      };
+
+      if ($scope.data.provisionOn == 'existingVms') {
+        resource.underline_provider_id = $scope.data.existingProviderId;
+      } else if ($scope.data.provisionOn == 'newVms') {
+        resource.underline_provider_id = $scope.data.newVmProviderId;
+        resource.masters_creation_template_id = $scope.data.masterCreationTemplateId;
+        resource.nodes_creation_template_id = $scope.data.nodeCreationTemplateId;
+        resource.master_base_name = $scope.data.createMasterBaseName;
+        resource.node_base_name = $scope.data.createNodesBaseName;
+      }
+      return resource;
+    };
+
+    $scope.ready = false;
+    var url = '/api/container_deployments/container_deployment_data';
+    API.get(url).then(function (response) {
+      'use strict';
+      $scope.deploymentData = response.data;
+      initializeDeploymentWizard();
+      $scope.ready = true;
+    });
+
+    $scope.data = {};
+    $scope.deployComplete = false;
+    $scope.deployInProgress = false;
+
+    var startDeploy = function () {
+      $scope.deployInProgress = true;
+      $timeout(function () {
+        var url = '/api/container_deployments';
+        var resource = create_deployment_resource();
+        API.post(url, {"action" : "create", "resource" :  resource}).then(function (response) {
+          'use strict';
+          $scope.deployInProgress = true;
+          var automationID = response
+        });
+
+        if ($scope.deployInProgress) {
+          $scope.deployInProgress = false;
+          $scope.deployComplete = false;
+          $scope.deploySuccess = false;
+          $scope.deployFailed = false;
+          $scope.deployFailureMessage = __("An unknown error has occurred.");
+        }
+      }, 5000);
```

Comment:
Aaaah... you need to inject `API` into the controller...

Right now, you're using the global `API` variable, that provides API access to non-angular code and is not synced with the angular's `$digest` cycle. If you inject `API`, it will use the angular service, whcih is exactly the same, except triggers a `$digest` immediately after the response..